src/gridfs/download.rs

```diff
@@ -139,16 +142,206 @@ impl GridFsBucket {
             n += 1;
         }
 
-        let expected_n =
-            total_bytes_expected / chunk_size + u64::from(total_bytes_expected % chunk_size != 0);
-        if (n as u64) != expected_n {
+        if n != file.n() {
             return Err(ErrorKind::GridFs(GridFsErrorKind::WrongNumberOfChunks {
                 actual_number: n,
-                expected_number: expected_n as u32,
+                expected_number: file.n() as u32,
             })
             .into());
         }
 
         Ok(())
     }
 }
+
+pub struct GridFsDownloadStream {
+    state: State,
+    current_n: u32,
+    file: FilesCollectionDocument,
+}
+
+type GetBytesFuture = BoxFuture<'static, Result<(Vec<u8>, Box<Cursor<Chunk<'static>>>)>>;
```

Comment:
could you explain the usage of `'static`? my prior understanding was that static lifetimes were for things that would live for the lifetime of a program, but it sounds wrong to me that every `Chunk` we encounter would live for the lifetime of the program so I think I'm missing something 🙂

Reply:
ahhh this post is very helpful! thanks for the explanation. I think I at least sort of get it now 🙂